### PR TITLE
syncer: improve performance - part 2

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

improve binlog replication performance in some cases.

### What is changed and how it works?

- do not use `null` value as the concurrence dispatch key
- decrease  default value of `queue-size` from `5120` to `1024`
- add `checkpoint-flush-interval` config item
- update `binlogSkippedEventsTotal` to `skipBinlogDurationHistogram` to support observe duration and update the Grafana dashboard

image

image

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to update the documentation
 - Need to update the `dm/dm-ansible`
 - Need to be included in the release note

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@WangXiangUSTC @lichunzhu PTAL

File: syncer/dml.go [hunk 1]
```diff
@@ -443,20 +444,42 @@ func findColumns(columns []*column, indexColumns map[string][]string) map[string
...
+		}
 		// for key, I think no need to add the `,` separator.
 		buf.WriteString(columnValue(data, columns[i].unsigned, columns[i].tp))
 	}
+	if buf.Len() == 0 {
+		log.L().Debug("all value are nil, no key generated", zap.String("table", table))
+		return "" // all values are `null`.
+	}
+
+	buf.WriteString(table)
```
Comment (WangXiangUSTC) [inline syncer/dml.go:460]: why write table at the end?

File: syncer/syncer.go [hunk 2]
```diff
@@ -1454,7 +1455,7 @@ func (s *Syncer) handleRowsEvent(ev *replication.RowsEvent, ec eventContext) err
 		return err
 	}
 	if ignore {
-		binlogSkippedEventsTotal.WithLabelValues("rows", s.cfg.Name).Inc()
+		skipBinlogDurationHistogram.WithLabelValues("rows", s.cfg.Name).Observe(time.Since(ec.startTime).Seconds())
```
Comment (WangXiangUSTC) [inline syncer/syncer.go:1458]: what is this metirc used for?  it may cost a lot of time in this function?

Comment (csuzhangxc) [reply syncer/syncer.go:1458]: some users may need to ignore most of the data, you can take a look at our internal ONCALL-1243 issue.

File: syncer/dml.go (hunk 1 above)
Comment (csuzhangxc) [reply syncer/dml.go:460]: if all columns are `null`, the return `""` (without `table`).

**Comment by csuzhangxc** [comment]
> seems that we have too many debug log, if we are debug other issue and need to enable debug log, these logs may disturb us.

em... but these logs are still needed for _debug_ (or just let them as the placeholder). maybe when debugging, we can choose to?

1. change some of them to _info_ or higher
2. do not run with _debug_ too long

**Comment by WangXiangUSTC** [comment]
> > seems that we have too many debug log, if we are debug other issue and need to enable debug log, these logs may disturb us.
> 
> em... but these logs are still needed for _debug_ (or just let them as the placeholder). maybe when debugging, we can choose to?
> 
> 1. change some of them to _info_ or higher
> 2. do not run with _debug_ too long

OK，LGTM

**Pull request merged**